go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -259,6 +262,12 @@ def handle_subprocess_error(stderr)
               new(go_mod_path, match[1], match[2])
           end
 
+          # we explicitly don't raise an error for 'go mod tidy' and silently
+          # continue here. `go mod tidy` shouldn't block updating versions
+          # because there are some edge cases where it's ok to fail (such as
+          # generated files not available yet to us). 
+          return if istidy
```

Comment:
We still fail with `Dependabot::DependencyFileNotResolvable` and `Dependabot::GoModulePathMismatch` when running a `go mod tidy`. Initially I wanted to move up this line so it never fails, but it looks we have some test cases that cover these two specific error types. I know this you PR fixes one particular case, but wonder if we should also not take care of `Dependabot::DependencyFileNotResolvable` and `Dependabot::GoModulePathMismatch` ?

Reply:
We run `go get` ([here](https://github.com/dependabot/dependabot-core/blob/499bc1adfaa583971c17d8f87edef0bc8719a9bf/go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb#L84)) before we run `go mod tidy`. With that in mind, if there's something in the manifest that would trigger a `Dependabot::DependencyFileNotResolvable` error or a `Dependabot::GoModulePathMismatch` error, can we reasonably assume that those errors would have occurred during `go get`? If so, then we know that we won't even get to the `go mod tidy` step in that case. What do you think?

Reply:
That's a good point Jason.  `go get` would definitely catch some of these (especially the `Dependabot::DependencyFileNotResolvable` errors).